tests/integration/dataframe/test_dataframe.py

```diff
@@ -615,3 +628,217 @@ def test__export_table_data__works(self, client: DataFrameClient, create_table):
+        self, client: DataFrameClient
+    ):
+        pa = pytest.importorskip("pyarrow")
+        table_id = "mock_table_id"
+        bad_batch = pa.record_batch([pa.array([1, 2, 3])], names=["b"])
+        api_info_json = {
+            "operations": {
+                "create_tables": {"available": True, "version": 1},
+                "delete_tables": {"available": True, "version": 1},
+                "modify_metadata": {"available": True, "version": 1},
+                "list_tables": {"available": True, "version": 1},
+                "read_data": {"available": True, "version": 3},
+                "write_data": {"available": True, "version": 1},
+            }
+        }
+
+        with responses.RequestsMock() as rsps:
+            rsps.add(
+                responses.GET,
+                f"{client.session.base_url}",
+                json=api_info_json,
+            )
+            rsps.add(
+                responses.POST,
+                f"{client.session.base_url}tables/{table_id}/data",
+                status=400,
+            )
+            with pytest.raises(ApiException) as excinfo:
+                client.append_table_data(table_id, [bad_batch], end_of_data=True)
+
+        assert "Arrow ingestion request was rejected" in str(excinfo.value)
+
+    def test__append_table_data__arrow_ingestion_400_supported_passthrough(
+        self, client: DataFrameClient
+    ):
+        pa = pytest.importorskip("pyarrow")
+        table_id = "mock_table_id"
+        bad_batch = pa.record_batch([pa.array([1, 2, 3])], names=["b"])
+        api_info_json = {
+            "operations": {
+                "create_tables": {"available": True, "version": 1},
+                "delete_tables": {"available": True, "version": 1},
+                "modify_metadata": {"available": True, "version": 1},
+                "list_tables": {"available": True, "version": 1},
+                "read_data": {"available": True, "version": 3},
+                "write_data": {"available": True, "version": 2},
+            }
+        }
+
+        with responses.RequestsMock() as rsps:
+            rsps.add(
+                responses.GET,
+                f"{client.session.base_url}",
+                json=api_info_json,
+            )
+            rsps.add(
+                responses.POST,
+                f"{client.session.base_url}tables/{table_id}/data",
+                status=400,
+            )
+            with pytest.raises(ApiException) as excinfo:
+                client.append_table_data(table_id, [bad_batch], end_of_data=True)
+
+        assert "Arrow ingestion request was rejected" not in str(excinfo.value)
+
+    def test__append_table_data__arrow_ingestion_non_400_passthrough(
+        self, client: DataFrameClient
+    ):
+        pa = pytest.importorskip("pyarrow")
+        table_id = "mock_table_id"
+        batch = pa.record_batch([pa.array([1, 2, 3])], names=["a"])
+        with responses.RequestsMock() as rsps:
```

Comment:
I don't feel strongly about it, but you could easily get a 404 by passing a (valid) table ID that doesn't exist.